### PR TITLE
Remove the remnants of the selective reload listener

### PR DESCRIPTION
**Pull request by ichttt**
The selective reload logic has been dead since 1.14 (see https://github.com/MinecraftForge/MinecraftForge/issues/5837), and with each update, more and more patches for the logic have been dropped.

With 1.17.1, the ISelectiveResourceReloadListener got deleted as well, but the deprecation Marker still exists in the ResourceManagerReloadListener.

This PR removes all that is left of the original logic.
Nobody felt like reimplementing this since 1.14 introduced threaded resource loading, so I belive we should just kill it completly. It would probably need a rewrite anyways if somebody would try to reimplement the feature.

### CODE REVIEW

**Review by pupnewfster — COMMENTED** [review]
This looks relatively good and it will definitely be nice to not deprecation warnings when implementing `ResourceManagerReloadListener`. One other patch that should also probably be removed as part of this that I ran into when doing a brief search is [this](https://github.com/MinecraftForge/MinecraftForge/blob/b7379953d50a04f4cc7f7c9912969959ed6268d2/patches/minecraft/net/minecraft/client/Minecraft.java.patch#L150).

**Comment by LexManos** [comment]
Honestly, even tho 1.17 hasn't officially had a RB yet, this is just a lot of binary breaks for breaks sake.
Better to mark things as deprecated/remove the patches then to break everything as is.
Now with J16 we can have since/forRemoval in @Deprecated, probably worth adding a gradle build verifier to make sure we strip all those out in 1.18.

**Review by pupnewfster — COMMENTED** [review]
I think it looks good now, though the deprecated annotations should probably have a `since = "1.18"` as well given I know lex would like to eventually have a check get ran that ensures things are removed in the next MC version port when they are marked as deprecated/for removal.

Edit: 1.18 not 1.17, as the idea is to have it be the version it will get removed instead of actually when it was deprecated.

File: patches/minecraft/net/minecraft/server/packs/resources/ResourceManagerReloadListener.java.patch [hunk 1]
```diff
@@ -1,22 +1,12 @@
...
-    default CompletableFuture<Void> m_5540_(PreparableReloadListener.PreparationBarrier p_10752_, ResourceManager p_10753_, ProfilerFiller p_10754_, ProfilerFiller p_10755_, Executor p_10756_, Executor p_10757_) {
-       return p_10752_.m_6769_(Unit.INSTANCE).thenRunAsync(() -> {
-@@ -17,4 +_,9 @@
+@@ -17,4 +_,11 @@
     }
  
     void m_6213_(ResourceManager p_10758_);
 +
++   /** @deprecated Selective reloading is no longer supported due to vanilla now reloading resources in parallel, so this is no longer needed */
++   @Deprecated(forRemoval = true)
```
Comment (pupnewfster) [inline patches/minecraft/net/minecraft/server/packs/resources/ResourceManagerReloadListener.java.patch]: This is missing a `since="1.17.1"`

Comment (ichttt) [reply patches/minecraft/net/minecraft/server/packs/resources/ResourceManagerReloadListener.java.patch]: Shows how much out of the loop I am, I found this as well when checking what I needed to update but forgot to gen patches...

**Pull request merged**